### PR TITLE
Us127040/save categories

### PR DESCRIPTION
**Pull request by ronan-f**
https://rally1.rallydev.com/#/29180338367d/dashboard?detail=/userstory/521443615208

### CODE REVIEW

**Review by wongvincent — APPROVED** [review]
Are separate assignment PATCH requests combined by `siren-sdk` into one request?
I can't remember. I think it does that for the activity-usage ones.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js [hunk 1]
```diff
@@ -46,25 +46,48 @@ class AssignmentCategoriesEditor extends ActivityEditorMixin(RtlMixin(LocalizeAc
 			return html`<div class="d2l-body-compact">${name}</div>`;
 		}
 
+		const unselectedId = '0'; // API expects 0 to unselect an ID
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js:49]: Nice

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js [hunk 2]
```diff
@@ -46,25 +46,48 @@ class AssignmentCategoriesEditor extends ActivityEditorMixin(RtlMixin(LocalizeAc
...
 		return html`
 			<select
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js:51]: Does this handle read-only view when a user cannot change categories? Not sure which permission controls this.

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js [hunk 3]
```diff
@@ -46,25 +46,48 @@ class AssignmentCategoriesEditor extends ActivityEditorMixin(RtlMixin(LocalizeAc
...
 				class="d2l-input-select d2l-block-select"
 				@change="${this._updateCategory}">
```
Comment (mostlyfabulous) [inline components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js:54]: Do we need a `disabled` attribute when permissions are missing?

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js (hunk 2 above)
Comment (ronan-f) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js:51]: Yip if you check out line 44 in this file it does a canEditCategories check which is determined by the presence of the update action in the HM response

File: components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js (hunk 3 above)
Comment (ronan-f) [reply components/d2l-activity-editor/d2l-activity-assignment-editor/d2l-activity-assignment-categories-editor.js:54]: Lacking permissions just renders a readonly view

**Pull request merged**